### ISSUE TITLE
Extra characters while decoding warn and error log files

### ISSUE THREAD

**Issue by sridevi-15** [couchbaselabs/couchbase-mobile-tools#12]
**Version:**
cbl-log #100

Steps to reproduce:
1. run cbl-log decoder tool for warning and error binary files

Expected Result:
You should see the output of the file should get decoded to text format

Actual Result: 
See the log on both warn and error witch etrac characters 
^[[1m^[[31mWARNING^[[0m:  -> warn file
^[[1m^[[31mERROR^[[0m:  -> error file

logs: 
[xamain-error.log](https://github.com/couchbaselabs/couchbase-mobile-tools/files/3105210/xamain-error.log)
[xamarin-warn.txt](https://github.com/couchbaselabs/couchbase-mobile-tools/files/3105212/xamarin-warn.txt)

This issue is happening on all platforms of cbl-log decoder except on windows . 
Tested on Centos, macOs, ubuntu
Happening with all platforms of cbl warn and error logs, other logs looks fine

**Comment by borrrden** [comment]
Could you provide the original binary log too?  And do you mean that if you run the command on Windows that cbl-log produces different results?  "Other logs" means info verbose and debug?

**Comment by snej** [comment]
Those are ANSI escape characters for bold/color. This code is producing them:
```c++
    vector<string> kLevels = {"***", "", "",
        ansiBold() + ansiRed() + "WARNING" + ansiReset(),
        ansiBold() + ansiRed() + "ERROR" + ansiReset()};
```
But they're only supposed to be written if stdout is a tty that supports ANSI color.

Sridevi, how exactly did you run the tool? Was it in a terminal and was the output being redirected to a file or piped into something...?

Also, what Git commit does build 100 correspond to?

**Comment by borrrden** [comment]
> What git build

[From the manifest](https://latestbuilds.service.couchbase.com/builds/latestbuilds/couchbase-lite-log/2.5.0/100/couchbase-lite-log-2.5.0-100-manifest.xml) commit a0fb7627cc071e6ff9653b13a70629c735673337 of mobile-tools with commit 09e4c055573721d63a5e0cf1405d273a18072f11 of LiteCore

**Comment by borrrden** [comment]
From my analysis the tool only checks if standard out is a TTY.  That remains true with the second argument to logcat.

**Comment by borrrden** [comment]
I believe the fix should be to add a conditional to the lines @snej mentioned:

Ref: https://github.com/couchbaselabs/couchbase-mobile-tools/blob/a0fb7627cc071e6ff9653b13a70629c735673337/cbl-log/cbl-log.cc#L98-L100

If the outputPath is empty, then we are writing to `stdout` and can use the current logic, otherwise the levels need to contain just text and no ANSI codes.

**Comment by snej** [comment]
There's a second argument to logcat?

**Comment by snej** [comment]
Cherry-picked fix to `release/iridium` branch as ca2e45d821cadd144da84a97d1fc6705f4b3e07e.
